### ISSUE TITLE
"Unlisted" flag is repsected by Tables page, but not sidebar

### ISSUE THREAD

**Issue by benhaynes** [RNGR/Directus#817]
By making a table "unlisted" (1) within `directus_privileges`, it does remove it from the Tables listing page, but also needs to remove it from the sidebar navigation.

![screen shot 2015-06-08 at 12 01 53 pm](https://cloud.githubusercontent.com/assets/522079/8039236/61de9e9c-0dd6-11e5-93c3-a3132e3b4ed9.png)

**Comment by WellingGuzman** [comment]
@benhaynes now I'm wondering what's the different between `unlisted` and `hidden`?

**Comment by WellingGuzman** [comment]
`unlisted` is part of **groups/directus_privileges** and `hidden` part of **directus_tables**.

We can update the `hidden` property on the Table Settings, but we can't modify the `unlisted` property on User Permission settings.

**Comment by benhaynes** [comment]
`hidden` applies to the table system-wide... as for junction tables, they are completely hidden from the whole system. `unlisted` means that a specific group will not see that table listed in the sidebar or table-listing.

`unlisted` is more useful since it's controlled per user-group... we could keep that and remove `hidden` if you want. Then we can just hide anything flagged as  `is_junction_table`.

Thoughts?

**Comment by benhaynes** [comment]
@WellingGuzman Correct, I think we could remove the global `hidden` (directus_tables) and only use `hidden` (directus_privileges)

**Comment by WellingGuzman** [comment]
@benhaynes You are right about having `hidden` as a table system-wide, but is there some other situation where a table need to be hidden to everyone, besides all junction tables?

What if  it's installed into a shared database and there a bunch of unrelated tables, that where should `hidden` be handy? it wouldn't be to update each group to unlisted those tables.

**Comment by WellingGuzman** [comment]
**Side note:** Also we should filter all the tables in one place, the `SchemaManager` so it doesn't matter the place we are trying to pull the tables, the manager will filter there.

**Comment by WellingGuzman** [comment]
Also we have `tab_blacklist` (`nav_blacklist`) which also "unlisted" tables from the sidebar.

I call this, let me know what are you thoughts, having `hidden`, `unlisted`, and `nav_blacklist` as the following:

- `hidden` will hide a table system-wise. **NO** show, editing, listing, etc. any where. (or we can just make it work the same as all `directus_` tables.
- `unlisted` group-wise. same as above.
- `nav_blacklist` you want it to be listed but not in the sidebar, In case you want to use key tables or whatever.

And here we are  talking about "links" because you can access to the table, In order to do not let the user to view, edit the table, there a permission option for each.

Hope I made myself clear with the idea.

Any thoughts?

**Comment by benhaynes** [comment]
@WellingGuzman I agree with this thinking, and it'd be great to consolidate all of this into the `schemaManager` if possible. The only issue I see is that you say that `hidden` will not let that table be editable... but junction tables need to be editable (permission-wise) so that we can adjust relationships.

Also, `unlisted` would hide tables from the sidebar **and the _Table Listing_ page**.

When we resolve **[this ticket](https://github.com/RNGR/Directus/issues/818)** we'll have control over the system nav items (show_users, show_files, show_activity, show_messages). The original purpose of `nav_blacklist` was to manage _all_ nav items (Extensions, Tables, Users, Files, Activity, Messages)... but it's only managing Tables. We have `nav_override` which can be used to adjust the sidebar tables. The only thing not managed is `Extensions` – so maybe nav_blacklist becomes `extension_blacklist` (per group)?

**To summarize:**

* `hidden` – **system-level**, hides tables but allows for editing (need to edit junction tables)
* `unlisted` – **group-level**, hides tables from sidebar and table-listing (boolean column)
* `show_users`, `show_activity`, `show_messages`, and `show_files` – **group-level**, hides system nav items (boolean columns)
* `extension_blacklist` – **group-level**, hides extensions (CSV of extension IDs)
* `nav_override` – **group-level** allows custom order/groupings for tables in sidebar (stays the same)

**Comment by WellingGuzman** [comment]
> The original purpose of nav_blacklist was to manage all nav items (Extensions, Tables, Users, Files, Activity, Messages)... but it's only managing Tables.

I put _`example`_ in the `nav_blacklist` and it hid the Example extensions from the navigation so it's working for anything that is part of the navigation.

The rest is implemented Would you like to review it @benhaynes ?

**Comment by benhaynes** [comment]
That's great! 

When you say the rest is implemented, do you mean the "show_messages" fields too? I am not seeing those here or on the dev server (with git hooks to the dev branch):

https://github.com/RNGR/Directus/blob/development/api/schema.sql

**Comment by WellingGuzman** [comment]
@benhaynes by _the rest is implemented_ I mean I did everything but add `extension_blacklist` to the table, because we can hide extensions (including bookmarks and table) using `nav_blacklist`.

> I am not seeing those here or on the dev server (with git hooks to the dev branch):
>
> https://github.com/RNGR/Directus/blob/development/api/schema.sql

Oh right, I missed the `schema.sql`, it was all on migrations. I'm going to send a pull request. We also need to run migrations on git hooks as well, that's why is not on dev branch on dev server.

**Comment by benhaynes** [comment]
Ahh, that makes sense. Let me know when this is up and I'll review!

Thanks @WellingGuzman !

**Comment by WellingGuzman** [comment]
Did I tell you this was up? because it was up, my bad.

@benhaynes let me know if it work as it expected?

**Comment by benhaynes** [comment]
Looks great!!
